.clang-format

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: GPL-2.0
+#
+# clang-format configuration file. Intended for clang-format >= 11.
+#
+# For more information, see:
+#
+#   Documentation/process/clang-format.rst
+#   https://clang.llvm.org/docs/ClangFormat.html
+#   https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+#
+# Obtained from: https://github.com/torvalds/linux/blob/master/.clang-format
+---
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlines: Left
+AlignOperands: true
+AlignTrailingComments: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: false
+BinPackArguments: true
+BinPackParameters: true
```

Comment:
```suggestion
BinPackParameters: true
BitFieldColonSpacing: None
```

`clang-format` also seems to move bitfields components onto a single line, but I've not found how to disable that :disappointed:.

We're also getting spaces before colons for labels, but I'm not sure how to disable that, either.

Reply:
Can you point me at a file where you see this?

Reply:
- For bitfields: look for `[^ ]:[^ ]`, first example I see is in `struct policy_entry` in `bpf/lib/common.h`.

- For the space between label and colon: `skip_egress_gateway` in `bpf/bpf_lxc.c` is the first example I could find.

Reply:
Label space:  that seems to be controlled by `SpaceBeforeInheritanceColon` - which seems like a bug, but setting `SpaceBeforeInheritanceColon: false` is probably fine for us given that we don't (afaik) use any class inheritance in bpf code.